### ISSUE TITLE
An ISel failure related to the builder type

### ISSUE THREAD

**Issue by mskvortsov** [tonlabs/TON-Compiler#50]
```
target datalayout = "E-S257-i1:257:257-i8:257:257-i16:257:257-i32:257:257-i64:257:257-i257:257:257-p:257:257-a:257:257"
target triple = "tvm"

define void @foo(builder* %builder) {
entry:
  %b = load builder, builder* %builder, align 1
  %b_prime = call builder @llvm.tvm.stu(i257 1, builder %b, i257 1)
  store builder %b_prime, builder* %builder, align 1
  ret void
}

declare builder @llvm.tvm.stu(i257, builder, i257)
```
```
LLVM ERROR: Cannot select: t13: ch = store<(store 1 into %ir.builder)> t9:1, t12, t2, undef:i257
  t12: TVMBuilder = llvm.tvm.stu TargetConstant:i257<5491>, Constant:i257<1>, t9, Constant:i257<1>
    t10: i257 = TargetConstant<5491>
    t11: i257 = Constant<1>
    t9: TVMBuilder,ch = load<(load 1 from %ir.builder)> t7, t2, undef:i257
      t2: i257 = TVMISD::ARGUMENT TargetConstant:i257<0>
        t1: i257 = TargetConstant<0>
      t8: i257 = undef
    t11: i257 = Constant<1>
  t2: i257 = TVMISD::ARGUMENT TargetConstant:i257<0>
    t1: i257 = TargetConstant<0>
  t8: i257 = undef
In function: foo
```

**Comment by mskvortsov** [comment]
This is an extraction from failing smart-contract001.c in the CI run for #49.

**Comment by mskvortsov** [comment]
Resolved in #49.
